pkg/apiserver/certificate/certificate_test.go

```diff
@@ -161,6 +187,25 @@ func TestApplyServerCert(t *testing.T) {
 				}
 				return
 			}
+
+			if tt.selfSignedCert {
+				oldCertKeyContent := got.getCertificate()
+				err := wait.Poll(time.Second, 8*time.Second, func() (bool, error) {
+					newCertKeyContent := got.getCertificate()
+					equal := bytes.Equal(oldCertKeyContent, newCertKeyContent)
+					if tt.testRotate {
+						return !equal, nil
+					}
+					return equal, nil
+				})
+
+				failureMessage := "CA cert updated"
+				if tt.testRotate {
+					failureMessage = "CA cert not updated"
+				}
+				assert.Nil(t, err, failureMessage)
+			}
```

Comment:
So for tests which do not force a rotation, we just end up waiting one second (because `Poll` is used, and it will not run the function until after the first time interval). Do you think this adds any value, or we should just have this whole code block only for tests with rotation: `if tt.selfSignedCert && tt.testRotate { ...`, which would make tests a bit shorter and this test code a bit simpler?

Reply:
With Poll I do not believe it adds value. The original intention was to check that rotation does not occur. Ill remove it.